---
identifier: "CVE-2021-37658"
package_slug: "pypi/tensorflow"
title: "Access of Uninitialized Pointer"
description: "TensorFlow is an end-to-end open source platform for machine learning.
  The [implementation](https://github.com/tensorflow/tensorflow/blob/84d053187cb80d975ef2b9684d4b61981bca0c41/tensorflow/core/kernels/linalg/matrix_diag_op.cc)
  has incomplete validation that the value of `k` is a valid tensor. We have check
  that this value is either a scalar or a vector, but there is no check for the number
  of elements. If this is an empty tensor, then code that accesses the first element
  of the tensor is wrong. We have patched the issue in GitHub commit ff8894044dfae5568ecbf2ed514c1a37dc394f1b.
  The fix will be included in TensorFlow We will also cherrypick this commit on TensorFlow,
  TensorFlow, and TensorFlow, as these are also affected and still in supported range."
date: "2021-08-18"
pubdate: "2021-08-12"
affected_range: ">=2.3.0,<2.3.4||>=2.4.0,<2.4.3||>=2.5.0,<=2.6.0"
fixed_versions:
- "2.3.4"
- "2.4.3"
affected_versions: "All versions starting from 2.3.0 before 2.3.4, all versions starting
  from 2.4.0 before 2.4.3, all versions starting from 2.5.0 up to 2.6.0"
not_impacted: "All versions before 2.3.0, all versions starting from 2.3.4 before
  2.4.0, all versions starting from 2.4.3 before 2.5.0, all versions after 2.6.0"
solution: "Upgrade to versions 2.3.4, 2.4.3 or above."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2021-37658"
cvss_v2: "AV:L/AC:L/Au:N/C:P/I:P/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
uuid: "c3ddb7f0-d719-4adc-a43a-7374693a28d9"
